### PR TITLE
Added context to links in header and footer 

### LINKED ISSUE #1327: Link text is unclear
The text "Learn" in the nav-bar and the footer is unclear without context and may be confusing to screen readers.  You can add a special screen reader text than can help with accessibility.

Relevant code:
```
<a href="/learn/">Learn</a>
```

### PR DESCRIPTION
**Pull request by Srinithyee**
# Issue Description
Fixes #1327 
The text "Learn" in the nav-bar and the footer is unclear without context and may be confusing to screen readers. You can add a special screen reader text than can help with accessibility.
 If there's alt text that can help provide context for screen readers it will make the link more accessible.

## Summary of the issue.
The current code is not friendly to screen readers. The texts in the nav-bar and footer does not have context. 

## Changes Made
Although the issue focused on "Learn", I've fixed the other texts in the nav-bar and footer as well.
* **I have added relevant alt text to each text in the nav-bar and footer**

### CODE REVIEW

**Comment by nibble0101** [comment]
Hi @Srinithyee. Thanks for the PR.

I think it would be better if you provided a link to the corresponding issue/issues in the body of the PR rather than in the header. You can do it by using the `Fixes` keyword followed by `#` character and the issue number. For example  `Fixes #1327` will automatically link to issue `1327`. 

About the changes you have submitted, I noticed the  `alt` attributes are  on `a` tags. In the final generated HTML, will the `alt` attribute still be on `a` tags or it will be on `img` tags?  I am asking because as far as I know, `alt` is not valid as HTML attribute for `a` tag.  You can check the MDN documentation [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).

**Comment by Srinithyee** [comment]
> Hi @Srinithyee. Thanks for the PR.
> 
> I think it would be better if you provided a link to the corresponding issue/issues in the body of the PR rather than in the header. You can do it by using the `Fixes` keyword followed by `#` character and the issue number. For example `Fixes #1327` will automatically link to issue `1327`.
> 
I've made the asked changes. Do check if this is okay now. Thanks :)
> About the changes you have submitted, I noticed the `alt` attributes are on `a` tags. In the final generated HTML, will the `alt` attribute still be on `a` tags or it will be on `img` tags? I am asking because as far as I know, `alt` is not valid as HTML attribute for `a` tag. You can check the MDN documentation [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).
>
Ah yes, looks like it won't be generated. I think I could use the 'title' attribute instead. I'll make the changes and resubmit. Thanks for the review :)

**Comment by Srinithyee** [comment]
@nibble0101 
I've replaced 'alt' attribute with 'title'. Hope this should work as expected!

**Comment by nibble0101** [comment]
Thanks @Srinithyee . I personally don't know any other way of improving the accessibility in this situation than making the text 
 within the  `a` tag more descriptive. Probably something like `Learn Ocaml` which in this case may not be possible  because we are dealing with a menu item. Using `title` attribute surely does help  but it also has its own shortcomings as described [here](https://www.a11yproject.com/posts/2019-02-15-creating-valid-and-accessible-links/#toc_What-about-the-title-attribute-). Let us wait and see what the other reviewers think.

**Comment by Srinithyee** [comment]
> Thanks @Srinithyee . I personally don't know any other way of improving the accessibility in this situation than making the text
> within the `a` tag more descriptive. Probably something like `Learn Ocaml` which in this case may not be possible because we are dealing with a menu item. Using `title` attribute surely does help but it also has its own shortcomings as described [here](https://www.a11yproject.com/posts/2019-02-15-creating-valid-and-accessible-links/#toc_What-about-the-title-attribute-). Let us wait and see what the other reviewers think.

 Yes, I don't know any other way to make the link text clear either, we could wait for the other reviewers to give their opinions. But since @gs0510 opened the issue, I think we could ask her if this was what she was looking for or if she was expecting something more than this. I could then work on it, based on her suggestions.

**Comment by gs0510** [comment]
Thanks for your PR @Srinithyee! @nibble0101  has a very good point wrt title, which I missed earlier. I think renaming the navbar items is not going to possible, so we don't really have an alternative here other than to use title. I think we can keep this change, until we come across a better alternative.

File: template/footer.mpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 </div>
 <div class="column">
   <div class="entry">
-    <h1><a href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></h1>
+    <h1><a title="Official Documentation for Ocaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></h1>
```
Comment (avsm) [inline template/footer.mpp]:
```suggestion
    <h1><a title="Official Documentation for OCaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></h1>
```

correcting the spelling of "OCaml"

File: template/navbar.mpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-              <li ((! ifdef learn class="active" !))><a href="{{! cmd script/link_of_lang ((! get filename !)) "/learn/" !}}">{{! cmd script/translate ((! get filename !)) Learn !}}</a></li>
-              <li ((! ifdef documentation class="active" !))><a href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></li>
-              <li ((! ifdef packages class="active" !))><a href="https://opam.ocaml.org/">{{! cmd script/translate ((! get filename !)) Packages !}}</a></li>
-              <li ((! ifdef community class="active" !))><a href="{{! cmd script/link_of_lang ((! get filename !)) "/community/" !}}">{{! cmd script/translate ((! get filename !)) Community !}}</a></li>
-              <li ((! ifdef news class="active" !))><a href="{{! cmd script/link_of_lang ((! get filename !)) "/community/planet/" !}}">{{! cmd script/translate ((! get filename !)) News !}}</a></li>
+              <li ((! ifdef learn class="active" !))><a title=" Beginner's guide to OCaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/learn/" !}}">{{! cmd script/translate ((! get filename !)) Learn !}}</a></li>
```
Comment (avsm) [inline template/navbar.mpp]:
```suggestion
              <li ((! ifdef learn class="active" !))><a title="Beginner's guide to OCaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/learn/" !}}">{{! cmd script/translate ((! get filename !)) Learn !}}</a></li>
```

Removing an errant space from the title

File: template/navbar.mpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+              <li ((! ifdef documentation class="active" !))><a title="Official Documentation for Ocaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></li>
```
Comment (avsm) [inline template/navbar.mpp]:
```suggestion
              <li ((! ifdef documentation class="active" !))><a title="Official Documentation for OCaml" href="{{! cmd script/link_of_lang ((! get filename !)) "/docs/" !}}">{{! cmd script/translate ((! get filename !)) Documentation !}}</a></li>
```

**Comment by avsm** [comment]
Pushed to staging.ocaml.org to verify the titles work (they do!), and merging with fixes to minor typos. Thanks @Srinithyee for the contribution and @nibble0101 and @gs0510 for the reviews.

**Pull request merged**